server/src/utilities/generateComponent/helpers/generateNgComponent.ts

```diff
@@ -0,0 +1,97 @@
+import { readFileSync } from 'fs';
+import { join } from 'path';
+import { ClassDeclaration, Node, SyntaxKind, SourceFile } from 'ts-morph';
+
+import MezzuriteComponent from '../../../models/mezzuriteComponent';
+
+function generateNgComponent (filePath: string, sourceFile: SourceFile): MezzuriteComponent {
+  let component = null;
+
+  if (filePath != null && sourceFile != null) {
+    // TODO: Handle multiple classes in a file.
+    const sourceClass = sourceFile.getClasses()[0];
+
+    if (sourceClass != null) {
+      const hasMezzuriteDirective = containsMezzuriteDirective(filePath, sourceClass);
+      const name = sourceClass.getName();
+
+      component = {
+        checks: {
+          hasMezzuriteDirective
+        },
+        filePath,
+        name,
+        type: 'ngComponent'
+      };
+    }
+  }
+
+  return component;
+}
+
+function containsMezzuriteDirective (filePath: string, sourceClass: ClassDeclaration): boolean {
```

Comment:
Could you pls break this function into multiple.

Reply:
Resolved via https://github.com/Microsoft/vscode-mezzurite/pull/29/commits/f05af0e98e2981a86e6f9a19b6f00ecebdba44a5.